### PR TITLE
fix: fix meta revert bug when clustering compaction task fail

### PR DESCRIPTION
**Pull request by wayblink**
#34603

### CODE REVIEW

File: internal/datacoord/compaction_task_clustering.go [hunk 1]
```diff
@@ -343,11 +343,18 @@ func (t *clusteringCompactionTask) resetSegmentCompacting() {
 
 func (t *clusteringCompactionTask) processFailedOrTimeout() error {
 	log.Info("clean task", zap.Int64("triggerID", t.GetTriggerID()), zap.Int64("planID", t.GetPlanID()), zap.String("state", t.GetState().String()))
-	// revert segment level
+	// revert segments meta
 	var operators []UpdateOperator
+	// revert level of input segments
+	// L1 : L1 ->(processPipelining)-> L2 ->(processFailedOrTimeout)-> L1
+	// L2 : L2 ->(processPipelining)-> L2 ->(processFailedOrTimeout)-> L2
 	for _, segID := range t.InputSegments {
```
Comment (MrPresent-Han) [inline internal/datacoord/compaction_task_clustering.go:351]: A little bit confused, 
Why input segments will be marked as L2 level in the process of Executing. Will this make these L1 segments invisible in the process of clustering compaction?

File: internal/datacoord/compaction_task_clustering.go [hunk 2]
```diff
@@ -343,11 +343,18 @@ func (t *clusteringCompactionTask) resetSegmentCompacting() {
...
 		operators = append(operators, RevertSegmentLevelOperator(segID))
-		operators = append(operators, RevertSegmentPartitionStatsVersionOperator(segID))
+	}
+	// if result segments are generated but task fail in the other steps, mark them as L1 segments without partitions stats
+	for _, segID := range t.ResultSegments {
+		operators = append(operators, UpdateSegmentLevelOperator(segID, datapb.SegmentLevel_L1))
+		operators = append(operators, UpdateSegmentPartitionStatsVersionOperator(segID, 0))
```
Comment (MrPresent-Han) [inline internal/datacoord/compaction_task_clustering.go:357]: Will these reverted L1 sements be garbage collected then?

**Review by MrPresent-Han — COMMENTED** [review]
a little confused, hope to be explained

File: internal/datacoord/compaction_task_clustering.go (hunk 1 above)
Comment (wayblink) [reply internal/datacoord/compaction_task_clustering.go:351]: Actually it make a strong protection mix compaction won't include these L1 segments. It is still visible. Yes, I agree it is a little confused.

File: internal/datacoord/compaction_task_clustering.go (hunk 2 above)
Comment (wayblink) [reply internal/datacoord/compaction_task_clustering.go:357]: If the compaction stage is done, failed in the following steps. We accept this compaction. Mark the output L1. Next major compaction will regard them as L1.

**Pull request merged**